src/napari_builtins/io/_read.py

```diff
@@ -485,3 +486,78 @@ def napari_get_reader(
         return _csv_reader
 
     return _magic_imreader
+
+
+@contextmanager
+def _patch_viewer_new():
+    """Context manager to patch the viewer's new method."""
+    from napari.viewer import Viewer, current_viewer
+
+    original_new = Viewer.__new__
+    original_init = Viewer.__init__
+
+    def patched_init(self, *args, **kwargs):
+        Viewer.__init__ = original_init
+
+    def patched_new(cls, *args, **kwargs):
+        ndisplay = None
+        if len(kwargs) == 1 and 'ndisplay' in kwargs:
+            ndisplay = kwargs.pop('ndisplay')
+
+        if not kwargs and not args:
+            viewer = current_viewer()
+            if ndisplay is not None:
+                viewer.dims.ndisplay = ndisplay
+            if viewer is not None:
+                Viewer.__new__ = original_new
+                return viewer
+        Viewer.__init__ = original_init
+        return original_new(cls)
+
+    Viewer.__new__ = patched_new
+    Viewer.__init__ = patched_init
+    try:
+        yield
+    finally:
+        Viewer.__new__ = original_new
+        Viewer.__init__ = original_init
+
+
+def load_and_execute_python_code(path: str) -> list['LayerData']:
+    """Load and execute Python code from a file.
+
+    Parameters
+    ----------
+    path : str
+        Path to the Python file to be executed.
+    """
+    code = Path(path).read_text()
+    with _patch_viewer_new():
+        try:
+            exec(code)
```

Comment:
```suggestion
    script_env = {}
    with _patch_viewer_new():
        try:
            exec(code, script_env)
            globals().update(script_env)
```
What about something like this to make the viewer aware of the stuff defined in the script?
This fixes the keybind example and parameter sweep, but I'm not sure how safe it really is.

I thought this might push things to the console too, but it doesn't. That can be a followup.

Reply:
Using globals is bad decision as it is module namespace. I use a different approach to keep references alive. 

We need to directly transfer references to the console.

Reply:
Maybe this helps:
https://github.com/napari/napari/pull/8135#discussion_r2234163824
moving stuff to the console worked for me there.

Reply:
The globals part isn't needed I don't think.